### PR TITLE
Fix 'elf.h' file not found on macOS 10.15

### PR DESCRIPTION
**Pull request by omajid**
See: #1744

This is a backport of the fix that was merged into runtime for master: https://github.com/dotnet/runtime/pull/42756

### CODE REVIEW

**Comment by omajid** [comment]
cc @janvorli @asbjornu @dagood

**Comment by omajid** [comment]
This PR is against release/3.1. I have no plans to do this for 5.0, since I assume we can pick that up automatically when the original fix is merged into 5.0.

I suppose I can try to get this into coreclr's release/3.1 as well ...

**Comment by dagood** [comment]
> when the original runtime fix (currently only in `master`) is merged into runtime's release/5.0 branch.

Do you know if this is planned? I don't see a port PR (via PR threads or any backlinks in them).

**Review by dagood — APPROVED** [review]
Awesome, thank you. (And @janvorli for the change.)

**Comment by omajid** [comment]
> Do you know if this is planned?

The milestone on https://github.com/dotnet/runtime/pull/42756 is "5.0.0"... so I guess? Not sure.

**Review by crummel — APPROVED** [review]

**Comment by omajid** [comment]
Maybe we should close this PR and wait for https://github.com/dotnet/coreclr/pull/28106 to be merged into 3.1?

**Comment by janvorli** [comment]
@omajid it is not clear if it would be approved for 3.1 and even if it was approved, it would take some time before it gets merged. Usually the approved PRs to 3.1 are merged after the upcoming version of .NET Core 3.1 is released and then released in the version after that. So the soonest, it might get merged into release/3.1 branch after 3.1.9 is released.

**Comment by dagood** [comment]
A couple things changed since this all started:
1. The fix got more complicated.
2. It looks like it might not be ok for a homebrew package to reference a commit rather than a tag: https://github.com/Homebrew/homebrew-core/pull/60929#issuecomment-703541291.

The fix is still relatively small and still localized in a single file, and there's active work going on to port the fix to branches sooner rather than later, so I still don't object based on that.

But it would be better for you to take the patch as a homebrew patch! That way, you can stay on a tag, so you can unblock yourself without worrying about (2).

These two approaches are not mutually exclusive, so I'm going to merge now, however if you think you can get this working before November, I urge you to take the patch in homebrew rather than blocking yourself by depending on the tagless merge commit.

---

For reference:
`master`: https://github.com/dotnet/runtime/pull/42756
`release/5.0`: https://github.com/dotnet/runtime/pull/43031
`release/3.1`: https://github.com/dotnet/coreclr/pull/28106

**Pull request merged**
